src/Surface.cpp

```diff
@@ -843,11 +843,11 @@ int ZCylinder::intersection(Point* point, double angle, Point* points) {
      * Find the y where F(x0, y) = 0
      * Substitute x0 into F(x,y) and rearrange to put in
      * the form of the quadratic formula: ay^2 + by + c = 0 */
-    a = _B * _B;
+    a = 1.;
```

Comment:
Can you add a comment above that the equations below are simplified for the case of a cylinder aligned with the z-axis? this will be helpful if anyone ever looks at this and tries to map these equations back to the generic equations for a quadratic surface.